server/routes/auth.js

```diff
@@ -0,0 +1,37 @@
+const express = require("express");
+const router = express.Router();
+const { body } = require("express-validator");
+const User = require("../models/user");
+const authController = require("../controllers/auth-controller");
+
+router.put(
+  "/signup",
+  [
+    body("email")
+      .isEmail()
+      .withMessage("Please enter valid email")
+      .custom(async (value, { req }) => {
+        const userDoc = await User.findOne({ email: value });
+        if (userDoc) {
+          return Promise.reject("Email already exist");
+        }
+      })
+      .normalizeEmail(),
+    body("password").trim().isLength({ min: 8 }),
+    body("confirmPassword").custom((value, { req }) => {
+      if (value != req.body.password) {
+        throw new Error("Passwords do not match");
+      }
+      return true;
+    }),
+    body("name").trim().not().isEmpty(),
+  ],
+  authController.postSignup
+);
+
+// router.post('/login', [
+//     body("email")
+//       .isEmail()
+//       .withMessage("Please enter valid email"),
+// ] ,authController.login);
```

Comment:
We should not include unnecessary commented code in our commits

Reply:
I'll keep that in mind from next time